### PR TITLE
Some space fixes

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Fix summary for spaces
Handle non-default spaces in `catkin locate`

### CODE REVIEW

**Comment by mikepurvis** [comment]
Looks reasonable. Are there any other cases like this in the source, or is this the only one?

**Comment by MatthijsBurgh** [comment]
I checked the rest of `context`, which has most spaces stuff in it. Context has `package_SPACE_NAME_space` functions, but for devel and install space, this really different, so these are hard to generate dynamically.
For the other spaces it is just a path join. So we could skip if the attr already exists(devel and install), otherwise add the path join function.

As extending the space descriptions to add such custom functions is to complex IMO.

I also found that `catkin locate` can't handle other spaces.

**Comment by MatthijsBurgh** [comment]
Friendly ping ;)

**Comment by MatthijsBurgh** [comment]
@mikepurvis friendly ping ;)

**Comment by MatthijsBurgh** [comment]
Friendly ping @mikepurvis @timonegk ;)

File: catkin_tools/verbs/catkin_clean/cli.py [hunk 1]
```diff
@@ -191,21 +193,17 @@ def clean_profile(opts, profile):
...
-        if opts.devel and devel_exists:
-            spaces_to_clean_msgs.append(clr("[clean] Devel Space:   @{yf}{}").format(ctx.devel_space_abs))
-        if opts.install and install_exists:
-            spaces_to_clean_msgs.append(clr("[clean] Install Space: @{yf}{}").format(install_path))
+    if opts.spaces and not (opts.yes or opts.dry_run):
+        for space in opts.spaces:
+            if getattr(ctx, '{}_space_exists'.format(space))():
+                space_name = Context.SPACES[space]['space']
+                space_abs = getattr(ctx, '{}_space_abs'.format(space))
+                spaces_to_clean_msgs.append(clr("[clean] {}:     @{yf}{}").format(space_name, space_abs))
```
Comment (timonegk) [inline catkin_tools/verbs/catkin_clean/cli.py:206]:
```suggestion
                spaces_to_clean_msgs.append(clr("[clean] {:14} @{yf}{}").format(space_name + ':', space_abs))
```
I would suggest this change to align the output (as it was the case before).

**Review by timonegk — COMMENTED** [review]
Apart from a small cosmetic suggestion I think this looks fine.

**Comment by MatthijsBurgh** [comment]
When this PR is merged, could a new release be scheduled? As the current releases don't accept any non-default pluggable spaces during execution.

**Comment by timonegk** [comment]
Sure. I just released a new version (0.8.0) to PyPI. @wjwwood could you release the new version to the package repositories? This time I released from master and also tagged the version ;)

**Comment by wjwwood** [comment]
@timonegk sure thing, and thanks!

**Comment by wjwwood** [comment]
Ok, it is available from the ROS 1 repository and soon from the ROS 2 repository (waiting on build farm queue).

**Pull request merged**